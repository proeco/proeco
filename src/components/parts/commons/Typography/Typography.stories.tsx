import React from 'react';

import { Box } from '@mui/system';
import { Typography } from './Typography';

export default {
  title: 'parts/commons/Typography',
};

const Template = () => (
  <Box display="flex" flexDirection="column" gap="20px">
    <Typography variant="h1">h1: 48px</Typography>
    <Typography variant="h2">h2: 32px</Typography>
    <Typography variant="h3">h3: 24px</Typography>
    <Typography variant="h4">h4: 20px</Typography>
    <Typography variant="body1">body1: 16px</Typography>
    <Typography variant="body2">body2: 14px</Typography>
    <Typography variant="button">button: 14px</Typography>
    <Typography variant="caption">caption: 12px</Typography>
    <Typography variant="overline">overline: 10px</Typography>
  </Box>
);

export const Default = Template.bind({});
