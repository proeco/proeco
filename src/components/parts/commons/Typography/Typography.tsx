import React, { VFC, ComponentProps } from 'react';
import { Typography as MuiTypography } from '@mui/material';
import { styled } from '@mui/material/styles';

type Bold = {
  bold?: boolean;
  MaximumLines?: number;
};

type Props = ComponentProps<typeof MuiTypography> & Bold;

export const Typography: VFC<Props> = ({ bold, MaximumLines = 1, color = 'textColor.main', ...rest }) => {
  return <StyledTypography color={color} bold={bold ? 1 : 0} MaximumLines={MaximumLines} {...rest} />;
};

const StyledTypography = styled(MuiTypography)<{ bold: number; MaximumLines: number }>`
  text-transform: none;
  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
  display: -webkit-box;
  overflow: hidden;
  -webkit-line-clamp: ${(props) => props.MaximumLines};
  -webkit-box-orient: vertical;
`;
