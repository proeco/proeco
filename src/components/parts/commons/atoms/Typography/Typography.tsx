import React, { VFC, ComponentProps } from 'react';
import { Typography as MuiTypography } from '@mui/material';
import styled from 'styled-components';

type Bold = {
  bold?: boolean;
};

type Props = ComponentProps<typeof MuiTypography> & Bold;

export const Typography: VFC<Props> = ({ bold, ...rest }) => {
  return <StyledTypography bold={bold} {...rest} />;
};

const StyledTypography = styled(MuiTypography)<{ bold?: boolean }>`
  text-transform: none;
  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
`;
