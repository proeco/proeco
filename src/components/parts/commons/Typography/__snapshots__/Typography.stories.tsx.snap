// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots parts/commons/Typography Default Typography 1`] = `
<div
  className="MuiBox-root css-1v8my8o"
>
  <h1
    bold={0}
    className="MuiTypography-root MuiTypography-h1 css-h2n376-MuiTypography-root"
  >
    h1: 48px
  </h1>
  <h2
    bold={0}
    className="MuiTypography-root MuiTypography-h2 css-1c2ajmr-MuiTypography-root"
  >
    h2: 32px
  </h2>
  <h3
    bold={0}
    className="MuiTypography-root MuiTypography-h3 css-131ayj9-MuiTypography-root"
  >
    h3: 24px
  </h3>
  <h4
    bold={0}
    className="MuiTypography-root MuiTypography-h4 css-8gca48-MuiTypography-root"
  >
    h4: 20px
  </h4>
  <p
    bold={0}
    className="MuiTypography-root MuiTypography-body1 css-61ooy4-MuiTypography-root"
  >
    body1: 16px
  </p>
  <p
    bold={0}
    className="MuiTypography-root MuiTypography-body2 css-uziypf-MuiTypography-root"
  >
    body2: 14px
  </p>
  <span
    bold={0}
    className="MuiTypography-root MuiTypography-button css-nt9eod-MuiTypography-root"
  >
    button: 14px
  </span>
  <span
    bold={0}
    className="MuiTypography-root MuiTypography-caption css-1i7dtab-MuiTypography-root"
  >
    caption: 12px
  </span>
  <span
    bold={0}
    className="MuiTypography-root MuiTypography-overline css-ych3qs-MuiTypography-root"
  >
    overline: 10px
  </span>
</div>
`;

exports[`Storyshots parts/commons/Typography Long Text Typography 1`] = `
<div
  className="MuiBox-root css-1jixria"
>
  <h3
    bold={0}
    className="MuiTypography-root MuiTypography-h3 css-1no7jgx-MuiTypography-root"
    maximum_lines={2}
  >
    Proeco は、個人開発やチーム開発を応援するWEBアプリケーションです！
  </h3>
</div>
`;
